src/app/collective-rewards/rewards/hooks/useGetBackerRewardPerTokenPaid.ts

```diff
@@ -0,0 +1,59 @@
+import { useBuilderContext } from '@/app/collective-rewards/user'
+import { GaugeAbi } from '@/lib/abis/v2/GaugeAbi'
+import { AVERAGE_BLOCKTIME } from '@/lib/constants'
+import { useMemo } from 'react'
+import { Address } from 'viem'
+import { useReadContracts } from 'wagmi'
+
+const COINBASE_ADDRESS = '0xf7ab6cfaebbadfe8b5494022c4c6db776bd63b6b'
+
+export const useGetBackerRewardPerTokenPaid = (backer: Address, token: Address = COINBASE_ADDRESS) => {
+  const { data: builders, isLoading: isBuildersLoading, error: buildersError } = useBuilderContext()
+
+  const backerRewardPerTokenPaidCalls = builders
+    .filter(({ gauge }) => !!gauge)
+    .map(
+      ({ gauge }) =>
+        ({
+          address: gauge,
+          abi: GaugeAbi,
+          functionName: 'backerRewardPerTokenPaid',
+          args: [token, backer],
+        }) as const,
+    )
```

Comment:
Builders could potentially be quite large, so reducing iterations might be beneficial. If you don't agree, then please ignore :)


```suggestion
  const backerRewardPerTokenPaidCalls = builders
    .reduce((acc, {gauge}, i) => {
      if (gauge) {
        acc[i] = {
          address: gauge,
          abi: GaugeAbi,
          functionName: 'backerRewardPerTokenPaid',
          args: [token, backer],
        }
      }
      
      return acc
    }, [])
```

Reply:
I know we had a conversation on this, but since I don't think the Builders length isn't going to be problematic, I prefer to split the operations for readability (it may be subjective, I know 🙌🏽)